pkg/apis/infinispan/v1/types_util.go

```diff
@@ -592,13 +592,20 @@ func applyLabels(ispn *Infinispan, envvar, annotationName string) error {
 	err := json.Unmarshal([]byte(labels), &labelMap)
 	if err == nil {
 		if len(labelMap) > 0 {
-			svcLabels := ""
 			if ispn.Labels == nil {
 				ispn.Labels = make(map[string]string)
 			}
-			for name, value := range labelMap {
-				ispn.Labels[name] = value
-				svcLabels = svcLabels + name + ","
+			keys := make([]string, len(labelMap))
```

Comment:
@rigazilla can you create unit test for this functionality?

Reply:
@dmvolod let me see

Reply:
@dmvolod I've added some tests

Reply:
Thanks, @rigazilla 
Looks good, but CI seems to be hangs

Reply:
Linter was fails. Please verify